### ISSUE TITLE
0 bytes file when string replacement on stream

### ISSUE THREAD

**Issue by grassick** [lazd/gulp-replace#24]
This gives a 0 byte file:

```
source = require ('vinyl-source-stream');
browserify('./index.coffee', { extensions: ['.coffee'], basedir: './src/' }))
    .bundle()
    .on('error', gutil.log)
    .pipe(source('index.js'))
    .pipe(replace("CURRENT_VERSION", version))
    .pipe(gulp.dest('./dist/js/'))
```

Where this gives a correct replacement

```
makeBuffer = require('gulp-buffer');
source = require ('vinyl-source-stream');

browserify('./index.coffee', { extensions: ['.coffee'], basedir: './src/' }))
    .bundle()
    .on('error', gutil.log)
    .pipe(source('index.js'))
    .pipe(makeBuffer())
    .pipe(replace("CURRENT_VERSION", version))
    .pipe(gulp.dest('./dist/js/'))
```

Something going wrong with streams created by vinyl-source-stream?

**Comment by EvanCarroll** [comment]
Also seeing this.. Except mine hangs too.

**Comment by EvanCarroll** [comment]
I think this is related to this bug... https://github.com/rvagg/through2/issues/5

**Comment by EvanCarroll** [comment]
I went to fix this bug, and `through2` was removed as prereq. Good job on merging that patch set! Please push up to NPM. Ready to rock.

**Comment by lazd** [comment]
@EvanCarroll @grassick I've published [`gulp-replace@0.5.4`](https://github.com/lazd/gulp-replace/releases/tag/v0.5.4). Please let me know if this is fixed!
